resourcex/README.md

```diff
@@ -0,0 +1,243 @@
+# resourcex
+
+The `resourcex` package extends the `github.com/pulumi/pulumi/sdk/v3/go/common/resource` package with helpers
+for working with property values.
+
+1. `Extract` - Extract structured values from a property map, with tracking of unknownness and secretness.
```

Comment:
We just had a design meeting where we agreed on a different way to implement `Extract` (where outputs/secrets are encapsulated in `pulumix.OutputOf[T]`). Is this yet another system for working unmarshalling?

Do we also plan to maintain this, or will we deprecate it?

Reply:
I believe that we agreed in the meeting that `infer` is not ready to be adopted by the p/k provider, and that the provider should continue to use `resource.PropertyValue` as its primary abstraction. In keeping with that, this code is simply designed to provide views over `PropertyValue`.